test/segment-loader.test.js

```diff
@@ -1014,6 +1014,153 @@ function(assert) {
   assert.equal(loader.state, 'WAITING', 'waiting for response');
 });
 
+QUnit.test('processing segment reachable even after playlist update removes it',
+function(assert) {
+  let playlist = playlistWithDuration(40);
+
+  playlist.endList = false;
+
+  loader.playlist(playlist);
+  loader.mimeType(this.mimeType);
+  loader.load();
+
+  assert.equal(loader.state, 'WAITING', 'in waiting state');
+  assert.equal(loader.pendingSegment_.uri, '0.ts', 'first segment pending');
+
+  // wrap up the first request to set mediaIndex and start normal live streaming
+  this.requests[0].response = new Uint8Array(10).buffer;
+  this.requests.shift().respond(200, null, '');
+  mediaSource.sourceBuffers[0].buffered = videojs.createTimeRanges([[0, 10]]);
+  mediaSource.sourceBuffers[0].trigger('updateend');
+
+  assert.equal(loader.state, 'WAITING', 'in waiting state');
+  assert.equal(loader.pendingSegment_.uri, '1.ts', 'second segment pending');
```

Comment:
I think it would be good to also check `loader.pendingSegment_.segment.uri` is what you expect whenver checking `loader.pendingSegment_.uri` to to make sure the new segment object we are attaching to `segmentInfo` is the correct reference